src/bytes.rs

```diff
@@ -178,6 +230,27 @@ impl Bytes {
         }
     }
 
+    /// Creates a new `Bytes` from `BytesImpl` implementation.
+    ///
+    /// Useful if you want to construct `Bytes` from your own buffer implementation.
+    #[inline]
+    pub fn with_impl<T: BytesImpl>(bytes_impl: T) -> Bytes {
```

Comment:
I think the name`from_impl` might be more clear, since it consumes T and converts to `Bytes`.

Reply:
That makes sense. I agree on it.